src/control/jobs/Job.h

```diff
@@ -72,6 +72,12 @@ class Job {
      */
     virtual void afterRun();
 
+    /**
+     * Called when the job is to be removed from the
+     * scheduler.
+     */
+    virtual void onDelete();
+
```

Comment:
Is this necessary? It violates https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-ctor-virtual